app/src/main/java/org/mozilla/fenix/settings/Extensions.kt

```diff
@@ -4,26 +4,33 @@
 
 package org.mozilla.fenix.settings
 
-import android.view.View
 import android.widget.RadioButton
-import android.widget.TextView
 import androidx.appcompat.content.res.AppCompatResources
-import androidx.core.text.HtmlCompat
 import androidx.preference.Preference
 import mozilla.components.feature.sitepermissions.SitePermissions
 import mozilla.components.support.ktx.android.view.putCompoundDrawablesRelative
-import org.mozilla.fenix.R
 import org.mozilla.fenix.theme.ThemeManager
 
 fun SitePermissions.toggle(featurePhone: PhoneFeature): SitePermissions {
-    return when (featurePhone) {
-        PhoneFeature.CAMERA -> copy(camera = camera.toggle())
-        PhoneFeature.LOCATION -> copy(location = location.toggle())
-        PhoneFeature.MICROPHONE -> copy(microphone = microphone.toggle())
-        PhoneFeature.NOTIFICATION -> copy(notification = notification.toggle())
-        PhoneFeature.AUTOPLAY_AUDIBLE -> copy(autoplayAudible = autoplayAudible.toggle())
-        PhoneFeature.AUTOPLAY_INAUDIBLE -> copy(autoplayInaudible = autoplayInaudible.toggle())
-    }
+    return update(featurePhone, get(featurePhone))
```

Comment:
The previous implementation used to call `toggle`,  I'm not sure if missing something ? :)

Reply:
Fixed!

Reply:
🚀